src/client/refresh/soft/sw_main.c

```diff
@@ -1570,21 +1570,20 @@ GetRefAPI(refimport_t imp)
  * is only one software renderer.
  */
 
-static SDL_Window* window = NULL;
-static SDL_Surface *surface = NULL;
-static SDL_Texture *texture = NULL;
-static SDL_Renderer *renderer = NULL;
-static qboolean X11_active = false;
+static SDL_Window	*window = NULL;
+static SDL_Surface	*surface = NULL;
+static SDL_Texture	*texture = NULL;
+static SDL_Renderer	*renderer = NULL;
 
 /*
```

Comment:
why is this icon stuff here, this should already be done in glimp_sdl.c, called by `GLimp_InitGraphics()`

Reply:
The softrenderer doesn't use the APIs provided by glimp_sdl.c, it does it's own stuff.

Reply:
Would it be possible to refactor it so it does use the glimp_sdl.c stuff?
I don't think it's tied to OpenGL in any way.

Reply:
should be, maybe by add additional flag for create window without setting GL flags.

Reply:
This is already handled: `re.PrepareForWindow()` returns the `SDL_WINDOW_OPENGL` flag which is then used to create the window. Other rendering backends can just return 0 or whatever flag they need (`-1` means "an error happened", but that's not a valid SDL window flag)